src/targets/Logary.Targets.Stackdriver/Target_Stackdriver.fs

```diff
@@ -0,0 +1,173 @@
+  let json = Google.Protobuf.WellKnownTypes.Struct.Parser.ParseJson (Json.format <| Object overrides)
+  LogEntry(Severity = toSeverity m.level, JsonPayload = json)
+
+module internal Impl =
+
+  type State = {
+    logName : string
+    logger : LoggingServiceV2Client
+    resource : MonitoredResource
+  }
+
+  let mkLabels project resource =
+    match resource with
+    // check out https://cloud.google.com/logging/docs/api/v2/resource-list for the list of resources
+    // as well as required keys for each one
+    | ComputeInstance(zone, instance) -> [ "project_id", project
+                                           "zone", zone
+                                           "instanceId", instance ]|> dict
+    | Container(cluster, ns, instance, pod, container, zone) -> [ "project_id", project
+                                                                  "cluster_name", cluster
+                                                                  "namespace_id", ns
+                                                                  "instance_id", instance
+                                                                  "pod_id", pod
+                                                                  "container_name", container
+                                                                  "zone", zone ] |> dict
+  let resourceName = function
+  | ComputeInstance _ -> "gce_instance"
+  | Container _ -> "container"
+  
+  let mkMonitoredResource project resourceType =
+    let r = MonitoredResource(Type = resourceName resourceType)
+    r.Labels.Add(mkLabels project resourceType)
+    r
+
+  let writeRequests (conf : StackdriverConf) (state : State) (reqs : TargetMessage[]) =
+      let messages = ResizeArray<_>() // Updated by the loop.
+      let acks = ResizeArray<_>() // updated by the loop
+      
+      for m in reqs do
+        match m with
+        | Log (message, ack) ->
+          // Write and save the ack for later.
+          acks.Add ack
+          messages.Add <| write message
+        | Flush (ackCh, nack) -> ()
+
+      let writer = Job.Scheduler.isolate <| fun _ -> state.logger.WriteLogEntries(state.logName, state.resource, conf.labels, messages)   
+      let handler = Job.tryIn writer Job.result (Job.raises)
+      handler >>= fun res -> Job.conIgnore acks 
+  
+  let createState (conf : StackdriverConf) = 
+    { logger = LoggingServiceV2Client.Create()
+      resource = mkMonitoredResource conf.projectId conf.resource
+      logName = sprintf "projects/%s/logs/%s" conf.projectId conf.logId }
+
+  let dispose runtime state ackVar = 
+    // do! Job.Scheduler.isolate (fun _ -> (state :> IDisposable).Dispose())
+    ackVar *<= () :> Job<_>
+
+  // This is the main entry point of the target. It returns a Job<unit>
+  // and as such doesn't have side effects until the Job is started.
+  let loop (conf : StackdriverConf) // the conf is specific to your target
```

Comment:
Looking good. You *may* also choose the loop signature that supports last wills, *if* you want to, but this is absolutely fine, too.